### PR TITLE
Configurable TLS MinVersion and CipherSuites

### PR DESCRIPTION
**Pull request by Red-GV**
**What this PR does / why we need it**:
This PR updates the `weaveworks/common` and `grafana/dskit` libraries so that the `CipherSuites` and `MinVersion` fields in `crypto/tls` can be set for the various Loki components.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

Dependent on: 
- https://github.com/grafana/dskit/pull/217
- https://github.com/weaveworks/common/pull/256

**Checklist**
- [x] Reviewed the `CONTRIBUTING.md` guide
- [x] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: pkg/server/internal_server.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.Config.HTTPTLSConfig.TLSKeyPath, "internal-server.http-tls-key-path", "", "HTTP internal server key path.")
 	f.StringVar(&cfg.Config.HTTPTLSConfig.ClientAuth, "internal-server.http-tls-client-auth", "", "HTTP TLS Client Auth type.")
 	f.StringVar(&cfg.Config.HTTPTLSConfig.ClientCAs, "internal-server.http-tls-ca-path", "", "HTTP TLS Client CA path.")
+	f.StringVar(&cfg.Config.CipherSuites, "internal-server.http-tls-cipher-suites", "", "HTTP TLS Cipher Suites.")
+	f.StringVar(&cfg.Config.MinVersion, "internal-server.http-tls-min-version", "", "HTTP TLS Min Version.")
```
Comment (periklis) [inline pkg/server/internal_server.go:25]: Are there supposed to be two pairs of these flags?
```suggestion
	f.StringVar(&cfg.HTTPTLSConfig.CipherSuites, "internal-server.http-tls-cipher-suites", "", "HTTP TLS Cipher Suites.")
	f.StringVar(&cfg.HTTPTLSConfig.MinVersion, "internal-server.http-tls-min-version", "", "HTTP TLS Min Version.")
	
	f.StringVar(&cfg.GRPCTLSConfig.CipherSuites, "internal-server.http-tls-cipher-suites", "", "GRPC TLS Cipher Suites.")
	f.StringVar(&cfg.GRPCTLSConfig.MinVersion, "internal-server.http-tls-min-version", "", "GRPC TLS Min Version.")
```

Comment (periklis) [inline pkg/server/internal_server.go:25]: Are there supposed to be two pairs of these flags?
```suggestion
	f.StringVar(&cfg.HTTPTLSConfig.CipherSuites, "internal-server.http-tls-cipher-suites", "", "HTTP TLS Cipher Suites.")
	f.StringVar(&cfg.HTTPTLSConfig.MinVersion, "internal-server.http-tls-min-version", "", "HTTP TLS Min Version.")
	
	f.StringVar(&cfg.GRPCTLSConfig.CipherSuites, "internal-server.http-tls-cipher-suites", "", "GRPC TLS Cipher Suites.")
	f.StringVar(&cfg.GRPCTLSConfig.MinVersion, "internal-server.http-tls-min-version", "", "GRPC TLS Min Version.")
```

Comment (Red-GV) [reply pkg/server/internal_server.go:25]: Nope. The change in weaveworks applies the `CipherSuites` and `MinVersion` to both configurations if defined: https://github.com/weaveworks/common/pull/256

**Review by periklis — COMMENTED** [review]
Looks good to me

**Review by periklis — APPROVED** [review]

File: pkg/lokifrontend/frontend/v1/frontend.go [hunk 2]
```diff
@@ -118,6 +118,7 @@ func New(cfg Config, limits Limits, log log.Logger, registerer prometheus.Regist
 }
 
 func (f *Frontend) starting(ctx context.Context) error {
+	f.subservicesWatcher = services.NewFailureWatcher()
```
Comment (bboreham) [inline pkg/lokifrontend/frontend/v1/frontend.go:121]: How does this change in `Frontend` relate to the rest of the PR which is about TLS settings?

Comment (Red-GV) [reply pkg/lokifrontend/frontend/v1/frontend.go:121]: I added this in order to pass the test cases.

This change had to be made as a result of update `weaveworks/common`. In the newer version (which contained the TLS properties), an error/panic to occur when invoking the function without a watcher.

**Pull request merged**